test/unit/sdam/server_description.test.js

```diff
@@ -41,4 +41,10 @@ describe('ServerDescription', function() {
       });
     });
   });
+
+  it('should sensibly parse an ipv6 address', function() {
+    const description = new ServerDescription('abcd:f::abcd:abcd:abcd:abcd:27017');
```

Comment:
Per [rfc3986](https://www.ietf.org/rfc/rfc3986.txt) section 3.2.2, this is actually an invalid representation of the address. It must be represented in text as: `[abcd:f::abcd:abcd:abcd:abcd]:27017`. I'm trying to think of better ways for us parse these values, and wondering if we can re-use the url parser in node to handle these rules for us? Maybe if you just wrap the string in a dummy `http://${input}/` and parse that? The thing is that we're _already_ parsing from the `MongoClient` constructor, so we might be introducing new complexity just for testing purposes.

Reply:
We pull the IPv6 address out of the square brackets here 
https://github.com/mongodb/node-mongodb-native/blob/c8495e590c9a5ed4ff7e513376b92e677c61c801/src/connection_string.ts#L688 
and then reconstruct it as ip colon port here:
https://github.com/mongodb/node-mongodb-native/blob/c8495e590c9a5ed4ff7e513376b92e677c61c801/src/sdam/topology.ts#L154-L155
So this string matches what gets passed to the ServerDescription typically, I can try to change that behavior and work out what bugs pop up there? Or I can document this reason why this test has a poorly formatted address